org.scala-ide.sdt.core/src/org/scalaide/core/internal/decorators/markoccurrences/ScalaOccurrencesFinder.scala

```diff
@@ -24,28 +22,6 @@ case class Occurrences(name: String, locations: List[IRegion])
 class ScalaOccurrencesFinder(unit: InteractiveCompilationUnit) extends HasLogger {
   import ScalaOccurrencesFinder._
 
-  /* IMPORTANT:
-   * In the current implementation, all access to `indexCache` are confined to the Presentation
-   * Compiler thread (and this is why `indexCache` requires no synchronization policy).
-   */
-  private var indexCache: Option[TimestampedIndex] = None
-
-  private def getCachedIndex(lastModified: Long, currentCompiler: IScalaPresentationCompiler): Option[MarkOccurrencesIndex] = indexCache match {
-    case Some(TimestampedIndex(`lastModified`, WeakReference(index))) if index.global eq currentCompiler => Some(index)
```

Comment:
Hm, I don't really understand how the test is not enough to make sure the AST is up to date. If the cache is cleared early it may lead to some additional recompilations, but that shouldn't lead to anything being incorrect (as you noticed). If it's cache is hit, it means the source hasn't changed, so why wouldn't the AST be correct? That's the part I don't get.

Regarding performance issues, I prefer to be on the careful side when something is computed on every cursor move, click, or key pressed. I just pointed out how this was a serious problem years ago, leading to this cache, that you now removed. It's fine, but be aware and know where to look if things turn sour ;-) The ultimate test used to be `Typers.scala`, which at more than 5000 LOC, pushes the IDE to its limits. :)